### PR TITLE
Magento: allow subdirectory and create variable for bin/magento

### PR DESCRIPTION
**Pull request by schmengler**
so it can be overridden to a custom location

- [ ] Bug fix #…?
- [x] New feature?
- [ ] BC breaks?
- [ ] Tests added?
- [x] Docs added?

      Please, regenerate docs by running next command:
      $ php bin/docgen

### CODE REVIEW

**Comment by peterjaap** [comment]
In what situation would bin/magento be in a custom path?

**Comment by schmengler** [comment]
In combination with #3461 

maybe I should combine this into one PR @peterjaap

**Comment by Schrank** [comment]
Ah, here it is :D

**Comment by Schrank** [comment]
> In combination with https://github.com/deployphp/deployer/pull/3461

> maybe I should combine this into one PR @peterjaap

I'm fine having two and merge them together :-)

File: recipe/magento2.php [hunk 1]
```diff
@@ -75,16 +75,18 @@
     'var/view_preprocessed/*'
 ]);
 
+set('bin/magento', '{{release_or_current_path}}/bin/magento');
+
 set('magento_version', function () {
     // detect version
-    $versionOutput = run('{{bin/php}} {{release_or_current_path}}/bin/magento --version');
+    $versionOutput = run('{{bin/php}} {{release_or_current_path}}/{{bin/magento}} --version');
```
Comment (peterjaap) [inline recipe/magento2.php:82]: Isn't the `{{release_or_current_path}}` already added to `{{bin/magento}}` on line 78?

Comment (schmengler) [reply recipe/magento2.php:82]: Good catch, looks like I messed that up when resolving conflicts. I'll fix it

File: recipe/magento2.php [hunk 2]
```diff
@@ -40,51 +40,57 @@
...
-    'var/view_preprocessed/*'
+    '{{magento_dir}}/generated/*',
+    '{{magento_dir}}/pub/static/_cache/*',
+    '{{magento_dir}}/var/generation/*',
+    '{{magento_dir}}/var/cache/*',
+    '{{magento_dir}}/var/page_cache/*',
+    '{{magento_dir}}/var/view_preprocessed/*'
 ]);
 
+set('bin/magento', '{{release_or_current_path}}/{{magento_dir}}/bin/magento');
```
Comment (schmengler) [inline recipe/magento2.php:82]: @peterjaap Notable change: included {{magento_dir}} here

File: recipe/magento2.php [hunk 3]
```diff
@@ -124,12 +129,14 @@
 
 desc('Enables maintenance mode');
 task('magento:maintenance:enable', function () {
-    run("if [ -d $(echo {{current_path}}) ]; then {{bin/php}} {{current_path}}/bin/magento maintenance:enable; fi");
+    // do not use {{bin/magento}} because it would be in "release" but the maintenance mode must be set in "current"
```
Comment (schmengler) [inline recipe/magento2.php:132]: @peterjaap Notable change: when we force `current_path`, do not use `{{bin/magento}}`

**Review by peterjaap — APPROVED** [review]
LGTM!

**Pull request merged**